src/sdk/pynni/nni/compression/torch/pruning/auto_compress_pruner.py

```diff
@@ -0,0 +1,227 @@
+                Optional('op_names'): [str],
+            }], model, _logger)
+        elif self._base_algo in ['l1', 'l2']:
+            schema = CompressorSchema([{
+                'sparsity': And(float, lambda n: 0 < n < 1),
+                'op_types': ['Conv2d'],
+                Optional('op_names'): [str]
+            }], model, _logger)
+
+        schema.validate(config_list)
+
+    def calc_mask(self, wrapper, **kwargs):
+        return None
+
+    def compress(self):
+        """
+        Compress the model with AutoCompress.
+
+        Returns
+        -------
+        torch.nn.Module
+            model with specified modules compressed.
+        """
+        _logger.info('Starting AutoCompress pruning...')
+
+        sparsity_each_round = 1 - pow(1-self._sparsity, 1/self._num_iterations)
+
+        for i in range(self._num_iterations):
+            _logger.info('Pruning iteration: %d', i)
+            _logger.info('Target sparsity this round: %s',
+                         1-pow(1-sparsity_each_round, i+1))
+
+            # SimulatedAnnealingPruner
+            _logger.info(
+                'Generating sparsities with SimulatedAnnealingPruner...')
+            SApruner = SimulatedAnnealingPruner(
+                model=copy.deepcopy(self._model_to_prune),
+                config_list=[
+                    {"sparsity": sparsity_each_round, "op_types": ['Conv2d']}],
+                evaluator=self._evaluator,
+                optimize_mode=self._optimize_mode,
+                base_algo=self._base_algo,
+                start_temperature=self._start_temperature,
+                stop_temperature=self._stop_temperature,
+                cool_down_rate=self._cool_down_rate,
+                perturbation_magnitude=self._perturbation_magnitude,
+                experiment_data_dir=self._experiment_data_dir)
+            config_list = SApruner.compress(return_config_list=True)
+            _logger.info("Generated config_list : %s", config_list)
+
+            # ADMMPruner
+            _logger.info('Performing structured pruning with ADMMPruner...')
+            ADMMpruner = ADMMPruner(
+                model=copy.deepcopy(self._model_to_prune),
+                config_list=config_list,
+                trainer=self._trainer,
+                num_iterations=self._admm_num_iterations,
+                training_epochs=self._admm_training_epochs,
+                row=self._row,
+                base_algo=self._base_algo)
+            ADMMpruner.compress()
+
+            ADMMpruner.export_model(os.path.join(self._experiment_data_dir, 'model_admm_masked.pth'), os.path.join(
+                self._experiment_data_dir, 'mask.pth'))
+
+            # use speed up to prune the model before next iteration, because SimulatedAnnealingPruner & ADMMPruner don't take masked models
+            self._model_to_prune.load_state_dict(torch.load(os.path.join(
+                self._experiment_data_dir, 'model_admm_masked.pth')))
```

Comment:
Why reload the checkpoint?

Reply:
The model weights have changed after admm pruner.